Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/eventregistration/checkins/qrcode/DefaultQRCodeVerifierTest.kt

```diff
@@ -0,0 +1,111 @@
+package de.rki.coronawarnapp.eventregistration.checkins.qrcode
+
+import de.rki.coronawarnapp.environment.EnvironmentSetup
+import de.rki.coronawarnapp.util.TimeStamper
+import de.rki.coronawarnapp.util.security.SignatureValidation
+import io.kotest.assertions.throwables.shouldNotThrowAny
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.types.shouldBeInstanceOf
+import io.mockk.MockKAnnotations
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import kotlinx.coroutines.test.runBlockingTest
+import org.joda.time.Instant
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import testhelpers.BaseTestInstrumentation
+
+@RunWith(JUnit4::class)
+class DefaultQRCodeVerifierTest : BaseTestInstrumentation() {
+
+    @MockK lateinit var environmentSetup: EnvironmentSetup
+    @MockK lateinit var timeStamper: TimeStamper
+
+    private lateinit var qrCodeVerifier: QRCodeVerifier
+
+    @Before
+    fun setUp() {
+        MockKAnnotations.init(this)
+        every { environmentSetup.appConfigVerificationKey } returns PUB_KEY
+        qrCodeVerifier = DefaultQRCodeVerifier(
+            SignatureValidation(environmentSetup),
+            timeStamper
+        )
+    }
+
+    @Test
+    fun verifyEventSuccess() = runBlockingTest {
+        val time = 2687960 * 1_000L
+        every { timeStamper.nowUTC } returns Instant.ofEpochMilli(time)
+        shouldNotThrowAny {
+            val verifyResult = qrCodeVerifier.verify(ENCODED_EVENT)
+            verifyResult.shouldBeInstanceOf<QRCodeVerifyResult.Success>()
+            verifyResult.apply {
+                event.description shouldBe "CWA Launch Party"
+            }
+        }
+    }
+
+    @Test
+    fun verifyEventStartTimeWaning() = runBlockingTest {
```

Comment:
Maybe additional test cases would be useful to test if the `StartTimeWarning` and `EndTimeWarning` are not returned (for locations). This would test the complex `when` condition in `DefaultQRCodeVerifier` in more detail.

Reply:
I asked Max if he I get some singed location , otherwise I don't know how, can you provide signed locations ?

Reply:
same https://github.com/corona-warn-app/cwa-app-android/pull/2524#discussion_r590289679